plugins/rapid7_insight_agent/icon_rapid7_insight_agent/actions/get_all_agents_by_ip/action.py

```diff
@@ -0,0 +1,32 @@
+import insightconnect_plugin_runtime
+from .schema import GetAllAgentsByIpInput, GetAllAgentsByIpOutput, Input, Output, Component
+
+# Custom imports below
+from ipaddress import ip_address as IPAddress
+from insightconnect_plugin_runtime.exceptions import PluginException
+
+
+class GetAllAgentsByIp(insightconnect_plugin_runtime.Action):
+    def __init__(self):
+        super(self.__class__, self).__init__(
+            name="get_all_agents_by_ip",
+            description=Component.DESCRIPTION,
+            input=GetAllAgentsByIpInput(),
+            output=GetAllAgentsByIpOutput(),
+        )
+
+    def run(self, params={}):
+        # START INPUT BINDING - DO NOT REMOVE - ANY INPUTS BELOW WILL UPDATE WITH YOUR PLUGIN SPEC AFTER REGENERATION
+        ip_address = params.get(Input.IP_ADDRESS, "")
+        # END INPUT BINDING - DO NOT REMOVE
+
+        try:
+            IPAddress(ip_address)
+        except ValueError:
```

Comment:
The error message we get from this is really useful so we could use it below instead of duplicating the message for cause and assistance
```suggestion
        except ValueError as error:
```
```
ValueError: '192.168.0.1/255' does not appear to be an IPv4 or IPv6 address
```